include/umf/mempolicy.h

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#ifndef UMF_MEMPOLICY_H
+#define UMF_MEMPOLICY_H 1
+
+#include <umf/base.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct umf_mempolicy_t *umf_mempolicy_handle_t;
+
+typedef enum umf_mempolicy_membind_t {
+    UMF_MEMPOLICY_INTERLEAVE,
+    UMF_MEMPOLICY_BIND,
+    UMF_MEMPOLICY_PREFERRED
+} umf_mempolicy_membind_t;
+
+const umf_mempolicy_handle_t umfMempolicyInterleave(void);
+const umf_mempolicy_handle_t umfMempolicyBind(void);
+const umf_mempolicy_handle_t umfMempolicyPreferred(void);
```

Comment:
I understand that these functions return handles to predefined policy objects (no allocation inside these functions). 

But do we really need it? The `umfMempolicyCreate(UMF_MEMPOLICY_INTERLEAVE, policy)` can be used to get an interleaving policy with default parameters. Looks like a duplication that we will need to explain in the documentation.

@igchor @bratpiorka @pbalcer what do you think?

Reply:
as far i remember on the meeting we decided to keep those- but if you prefer to do not have those, i can remove it.  Others, what's your opinion?

Reply:
Yeah, that concern just came to my mind after I looked into this version of PR.

I have no strong opinion on that. Just prefer to avoid duplication.

Reply:
Numa nodes(memtargets) is defined in memspace. 
 
 You create pool from both structs.

Reply:
> Of course, right. But in general, we won't be able to set anything on the mempolicies, right? For example, if we want to add some access hint (read-only/read-write, etc.)

if you want set more, you use _create function, and then set options - const are for "fast path" for default use, so you dont have do extra code. 

let me show a difference. 
```c
    umf_memory_provider_handle_t hProvider = nullptr;

    umf_result_t ret = umfMemoryProviderCreateFromMemspace(
        hMemspace, umfMempolicyPreferred(), &hProvider);
    if (ret) {
         //error handling
    }
```

vs 
```c
    umf_memory_provider_handle_t hProvider = nullptr;
    umf_result_t ret = umfMempolicyCreate(UMF_MEMPOLICY_PREFERRED, &hPolicy);
    if (ret) {
        //error handling
    }
    ret = umfMemoryProviderCreateFromMemspace(
        hMemspace, umfMempolicyPreferred(), &hProvider);

    if (ret) {
        //error handling
    }
    ret = umfMempolicyDestroy(hPolicy);
    if (ret) {
        //error handling??
    }

```

Reply:
@lplewa I understand your point regarding the number of lines of code in both cases. With `umfMempolicyPreferred()` user required to write less lines of code.

But on the other hand, we should explain this to the user:  `if you want set more, you use _create function, and then set options - const is for "fast path" for default use`
My feeling suggests that fewer entry points are easier to explain to users.

Reply:
what do you think about umfMempolicyPreferred(umfMempolicyPreferredParamsDefault()) ?

Reply:
> what do you think about umfMempolicyPreferred(umfMempolicyPreferredParamsDefault()) ?

I did not get the point. How does it improve the situation?

Reply:
OK - after more thinking and conversation with Łukasz I like the option where:
* we don't introduce new umf_mempolicy_membind_t type and umfMempolicyCreate() accepts umf_const_mempolicy_handle_t, e.g. umfMempolicyCreate(umfMempolicyInterleaveGet(), ...)
* umfMempolicyCreate() could be changed to umfMempolicyDupicate() or similar

About adding an extra explanation - I think this is ok if we add additional comments in the header (for doxygen) since if we use const handles user gets a compilation error.